### PR TITLE
add pre-commit + flake8

### PR DESCRIPTION
**Pull request by gruebel**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

- added `pre-commit` + config + GitHub Action + docs
- added `flake8` + config + included to `pre-commit`
- fixed couple of issues, which were flagged by `flake8`

Step-by-step I will remove the ignored rules in the `flake8` config. Even for single rules I will split them up, because there too many small fixes and that way it is easier to review them and not get too overwhelmed 😱

### CODE REVIEW

File: checkov/common/checks_infra/checks_parser.py [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
+    LessThanAttributeSolver,
+    LessThanOrEqualAttributeSolver,
+)
 from checkov.common.graph.checks_infra.base_check import BaseGraphCheck
 from checkov.common.graph.checks_infra.base_parser import BaseGraphCheckParser
 from checkov.common.graph.checks_infra.enums import SolverType
 from checkov.common.graph.checks_infra.solvers.base_solver import BaseSolver
-from checkov.common.checks_infra.solvers import *
+
+# from checkov.common.checks_infra.solvers import *
```
Comment (schosterbarak) [inline checkov/common/checks_infra/checks_parser.py:36]: is this comment intended?

File: checkov/common/output/record.py [hunk 2]
```diff
@@ -89,7 +89,7 @@ def _code_line_string(code_block):
 
     def to_string(self, compact=False, use_bc_ids=False):
         status = ''
-        evaluation_message = f''
```
Comment (schosterbarak) [inline checkov/common/output/record.py:92]: is this breaking something?

File: checkov/common/checks_infra/checks_parser.py (hunk 1 above)
Comment (gruebel) [reply checkov/common/checks_infra/checks_parser.py:36]: 🙈 no, forgot to remove it

File: checkov/common/output/record.py (hunk 2 above)
Comment (gruebel) [reply checkov/common/output/record.py:92]: nope, the `f` before the string is not needed, because there is nothing to format 🙂

**Review by nimrodkor — APPROVED** [review]
Nice! @gruebel

**Pull request merged**